rotkehlchen/exchanges/htx.py

```diff
@@ -208,4 +346,84 @@ def query_online_trade_history(
             start_ts: Timestamp,
             end_ts: Timestamp,
     ) -> tuple[list[Trade], tuple[Timestamp, Timestamp]]:
-        return [], (start_ts, end_ts)  # TODO: Implement this in another PR
+        """
+        Query trades from HTX in the spot category.
+        The API is limited to query trades up to 120 days into the past. We can query time ranges
+        but we don't know when to stop querying into the past. What this function does is iterate
+        over the 120 days period moving the queried frame in the maximum allowed time span, that is
+        48 hours:
+        [end ts, end ts - 48h] -> [end ts - 48h, end ts - 96h] -> ... [end ts - n*48h, start ts]
+
+        Since we have a clear limit to stop querying what we do is use the start-time/end-time keys
+        to filter the data that we need.
+        """
+        new_trades: list[Trade] = []
+        upper_ts, lower_ts = end_ts, Timestamp(end_ts - DAY_IN_SECONDS * 2)
+        earliest_query_start_ts = Timestamp(ts_now() - DAY_IN_SECONDS * 120)
+
+        if end_ts <= earliest_query_start_ts:  # 0... start_ts ... end_ts ... earliest_query_start
+            return [], (start_ts, end_ts)  # entire query out of range. Bail
+
+        if start_ts <= earliest_query_start_ts:  # 0 ... start_ts ...  earliest_query_start ... end_ts  # noqa: E501
+            start_ts = Timestamp(earliest_query_start_ts + 60 * 5)  # margin of 5 minutes to avoid possible errors in case of delay  # noqa: E501
+
+        while True:
+            raw_data = self._query(
+                absolute_path='/v1/order/matchresults',
+                options={
+                    'start-time': str(ts_sec_to_ms(lower_ts)),
+                    'end-time': str(ts_sec_to_ms(upper_ts)),
+                    'size': 500,
+                },
+            )
+            for raw_trade in raw_data:
+                try:
+                    symbol, fee_currency = raw_trade['symbol'], raw_trade['fee-currency']
+                    trade_type = TradeType.deserialize(raw_trade['type'].split('-')[0])
+                    if trade_type == TradeType.BUY:
+                        base_asset, quote_asset = asset_from_htx(fee_currency), asset_from_htx(
+                            self.remove_fee_currency(symbol=symbol, fee_currency=fee_currency),
+                        )
+                        fee_asset = base_asset
+                    elif trade_type == TradeType.SELL:
+                        base_asset, quote_asset = asset_from_htx(
+                            self.remove_fee_currency(symbol=symbol, fee_currency=fee_currency),
+                        ), asset_from_htx(fee_currency)
+                        fee_asset = quote_asset
+                    else:
+                        raise UnprocessableTradePair(raw_trade['symbol'])
+                except (UnknownAsset, UnprocessableTradePair, KeyError):
+                    log.error(f'Could not read assets from HTX trade {raw_trade}')
```

Comment:
this is not the right message for the keyError and the other error. Please take care of the messages since they help us later to debug